packages/block-editor/src/components/inner-blocks/index.js

```diff
@@ -135,11 +142,14 @@ InnerBlocks = compose( [
 		} = select( 'core/block-editor' );
 		const { clientId } = ownProps;
 		const rootClientId = getBlockRootClientId( clientId );
+		const block = getBlock( clientId );
+
 		return {
 			isSelectedBlockInRoot: isBlockSelected( clientId ) || hasSelectedInnerBlock( clientId ),
-			block: getBlock( clientId ),
+			block,
 			blockListSettings: getBlockListSettings( clientId ),
 			parentLock: getTemplateLock( rootClientId ),
+			hasChildBlocks: !! ( block && block.innerBlocks.length ),
```

Comment:
I feel like for some blocks, this appender might be useful even if there are child blocks. For example in a column block (or similar)

Reply:
or even the section block. We could say, the appender is visible when the block is selected but hidden otherwise.

Reply:
It's true. I believe CoBlocks does something similar with the appender in the Columns Block.

I feel like this should be configurable but that might mean making the prop into an `Object` in order that we could pass through this config option:

```jsx
<InnerBlocks
    placeholderType={
    	type: 'block-appender',
    	options: {
    		'hideWhenChildren': true
    	}
  	}
/>
```

This might be getting a little messy though...

Advice appreciated.

Reply:
I personally quite like the idea of using an object. The example could be shortened to just:
```
<InnerBlocks
    appender={
    	type: 'button',
	'hideWhenChildren': true,
  }
/>
```

I can also see there being a desire for additional options, one that springs to mind is changing the button label.

Reply:
These proposals are reasonable. Another approach could be to do:

```
<InnerBlocks
    appender="button"
    showAppenderIfEmpty
> 
```
The advantage here is that we won't pass a new object instance on each render. 
The downsides is scaling (if we have more options), but it feels like these two options are distinct. One controls the appender UI, the other one controls a behavior of InnerBlocks instead.

pinging @aduth for thoughts?

Reply:
Testing it now with the section block, and I agree the appender should stay after the first block has been added. Without it, it's actually quite difficult to add another block afterwards.

Reply:
Hopefully, @aduth can recommend either

1. Separate props
2. Options object

> I agree the appender should stay after the first block has been added

This seems reasonable. Should be configurable though.

Reply:
>The downsides is scaling (if we have more options), but it feels like these two options are distinct. One controls the appender UI, the other one controls a behavior of InnerBlocks instead.

I agree with this assessment (in favor of individual props).

I'm also strongly inclined against using objects as prop values anyways, for difficulty in optimizing performance therein in creating new object references ([see existing non-ideal hacky workarounds for `layouts` prop](https://github.com/WordPress/gutenberg/blob/a6ff3a0622142bb5ace6de3cfc958966f5573951/packages/block-library/src/columns/utils.js#L7-L16)).

Reply:
> I feel like for some blocks, this appender might be useful even if there are child blocks. For example in a column block (or similar)

I've added a separate prop for render type selection and another to show/hide the appender depending on the presence of child blocks.